torch/csrc/autograd/input_metadata.h

```diff
@@ -11,22 +11,29 @@
 namespace torch { namespace autograd {
 
 /**
- * Records type, shape, and device of tensor and, where applicable,
- * the stream the correspondingoperation took place on.
+ * Records TensorOptions, and shape of the tensor and, where applicable,
+ * the stream the corresponding operation took place on.
  *
  * If is_valid() is false, then the corresponding input is not used and may be
  * an undefined tensor.
  */
 struct InputMetadata {
   InputMetadata() = default;
 
-  InputMetadata(const at::TensorOptions options, at::IntArrayRef shape, at::Device device)
-  : options_{options}, shape_{shape}, device_{device} {
+  InputMetadata(const at::TensorOptions options, at::IntArrayRef shape, bool is_tensor_subclass)
+  : options_{options}, shape_{shape}, is_tensor_subclass_{is_tensor_subclass} {
+    auto device_ = options.device();
+    stream_ = c10::impl::getDeviceGuardImpl(device_.type())->getStream(device_);
+  }
+
+  InputMetadata(const at::TensorOptions options, at::IntArrayRef shape)
```

Comment:
When is this constructor called?
How is the python dispatch initialized?

Reply:
removed it